pkg/client/connector/service.go

```diff
@@ -504,9 +505,16 @@ func run(c context.Context) error {
 		if !ok {
 			return nil
 		}
-		tm.sshPortForward(c,
-			"-D", "localhost:1080",
-		)
+		ln, err := net.Listen("tcp4", "127.0.0.1:0")
```

Comment:
I know it's in the function, but maybe a brief comment saying we are using this to randomly select an available port, just so people know why in the future.